diff --git a/Source/KNSoft.MakeLifeEasier/Process/Process.c b/Source/KNSoft.MakeLifeEasier/Process/Process.c
index 9b87613..c0f1759 100644
--- a/Source/KNSoft.MakeLifeEasier/Process/Process.c
+++ b/Source/KNSoft.MakeLifeEasier/Process/Process.c
@@ -11,24 +11,22 @@ PS_CreateProcess(
     _In_opt_ PCWSTR ApplicationName,
     _Inout_opt_ PWSTR CommandLine,
     _In_ LOGICAL InheritHandles,
+    _In_ ULONG CreationFlags,
     _In_opt_ PCWSTR CurrentDirectory,
     _In_opt_ LPSTARTUPINFOW StartupInfo,
     _Out_ LPPROCESS_INFORMATION ProcessInformation)
 {
     W32ERROR Ret;
-    PVOID Environment;
+    PVOID Environment = NULL;
     STARTUPINFOW si, *psi;
-    DWORD CreationFlags = NORMAL_PRIORITY_CLASS | CREATE_NEW_CONSOLE | CREATE_DEFAULT_ERROR_MODE;
 
     if (TokenHandle != NULL)
     {
-        if (CreateEnvironmentBlock(&Environment, TokenHandle, FALSE))
+        if (!CreateEnvironmentBlock(&Environment, TokenHandle, FALSE))
         {
-            CreationFlags |= CREATE_UNICODE_ENVIRONMENT;
+            return Err_GetLastError();
         }
-    } else
-    {
-        Environment = NULL;
+        CreationFlags |= CREATE_UNICODE_ENVIRONMENT;
     }
     if (StartupInfo != NULL)
     {
@@ -51,7 +49,7 @@ PS_CreateProcess(
                                  psi,
                                  ProcessInformation,
                                  NULL) ? ERROR_SUCCESS : Err_GetLastError();
-    if (Environment)
+    if (Environment != NULL)
     {
         DestroyEnvironmentBlock(Environment);
     }
diff --git a/Source/KNSoft.MakeLifeEasier/Process/Process.h b/Source/KNSoft.MakeLifeEasier/Process/Process.h
index 5d1c17e..05ca45f 100644
--- a/Source/KNSoft.MakeLifeEasier/Process/Process.h
+++ b/Source/KNSoft.MakeLifeEasier/Process/Process.h
@@ -22,6 +22,7 @@ PS_CreateProcess(
     _In_opt_ PCWSTR ApplicationName,
     _Inout_opt_ PWSTR CommandLine,
     _In_ LOGICAL InheritHandles,
+    _In_ ULONG CreationFlags,
     _In_opt_ PCWSTR CurrentDirectory,
     _In_opt_ LPSTARTUPINFOW StartupInfo,
     _Out_ LPPROCESS_INFORMATION ProcessInformation);
diff --git a/Source/Samples/CreateSuperToken/CreateSuperToken.c b/Source/Samples/CreateSuperToken/CreateSuperToken.c
index ee7a680..a3aa7d7 100644
--- a/Source/Samples/CreateSuperToken/CreateSuperToken.c
+++ b/Source/Samples/CreateSuperToken/CreateSuperToken.c
@@ -171,6 +171,7 @@ wmain(
                            CmdPath,
                            CmdPath,
                            FALSE,
+                           NORMAL_PRIORITY_CLASS | CREATE_NEW_CONSOLE | CREATE_DEFAULT_ERROR_MODE,
                            NULL,
                            NULL,
                            &ProcessInfo);
